composer/trainer/trainer.py

```diff
@@ -2591,6 +2593,8 @@ def eval(
                 subset_num_batches=subset_num_batches,
                 metrics=self.state.eval_metrics[evaluator.label],
             )
+            if eval_passed_in:
+                self.state.evaluators.remove(evaluator)  # Remove them from state once eval is finished.
```

Comment:
This removes all evaluators? but it should just remove the evaluator added

Reply:
In the case where `eval_dataloader` is passed in directly to `eval`, the `evaluators` variable only contains the passed in `eval_dataloader`s. Therefore it is safe to remove everything that is in the `evaluators` variable. Please double check this logic though.